### ISSUE TITLE
Age of Empire 4 : created image view with differing format from a swapchain image

### ISSUE THREAD

**Issue by djdeath** [HansKristian-Work/vkd3d-proton#1213]
Before the first image is visible on screen, we run into this assert in mesa : https://gitlab.freedesktop.org/mesa/mesa/-/blob/main/src/vulkan/runtime/vk_image.c#L418

This means that an image view was created out of an image but both formats different.

The image is actually coming from a swapchain. We'll set the `VK_IMAGE_CREATE_MUTABLE_FORMAT_BIT` if the swapchain is created with `VK_SWAPCHAIN_CREATE_MUTABLE_FORMAT_BIT_KHR` : https://gitlab.freedesktop.org/mesa/mesa/-/blob/main/src/vulkan/wsi/wsi_common.c#L477

So it seems vkd3d-proton is missing that flag when creating the swapchain.

**Comment by DadSchoorse** [comment]
This is probably actually a bug in proton's full screen emulation, I have a fix pending for it https://github.com/ValveSoftware/wine/pull/158

**Comment by doitsujin** [comment]
Yeah, this can't really be our bug, we explicitly **only** use the swap chain format itself to create image views and never forward the swap chain images directly to the application.

**Comment by renatopereyra** [comment]
> The question is whether or not we should work around this on our end.

Would a workaround still be needed given https://github.com/ValveSoftware/wine/pull/158?

**Comment by DadSchoorse** [comment]
@djdeath The fix should now be available in proton experimental.

**Comment by djdeath** [comment]
Cool, I guess we can close this then.
